lib/SIL/IR/SILInstruction.cpp

```diff
@@ -1519,6 +1519,21 @@ MultipleValueInstruction *MultipleValueInstructionResult::getParent() {
   return reinterpret_cast<MultipleValueInstruction *>(value);
 }
 
+/// Returns true if evaluation of this node may cause suspension of an
+/// async task.
+bool SILNode::maySuspend() const {
```

Comment:
Should this just be on `SILInstruction`?  Not sure there's any situation where it makes sense to ask this of a SILValue.

Reply:
Well, it doesn't seem harmful to put it on SILNode either, to save a cast if all you do have is a SILValue.

Reply:
I guess my question is whether it ever makes sense to just ask this of a SILValue.  It's never a property of a value, just of an instruction that can produce a value.  If you're asking a SILValue you're almost certainly doing it wrong.